### PR TITLE
Add AssemblyVersion attribute

### PR DESCRIPTION
**Pull request by 64J0**
## Description

With this PR, I'm adding the **AssemblyVersion** attribute to Giraffe.fsproj, and updating the DEVGUIDE document, adding a new step to update this attribute whenever a new release is intended.

## Related issues

- Close https://github.com/giraffe-fsharp/Giraffe/issues/521

### CODE REVIEW

**Comment by 64J0** [comment]
@baronfel and @DalekBaldwin, if you have some free time, please take a look at this PR. I think this must be enough to close the issue https://github.com/giraffe-fsharp/Giraffe/issues/521, and later we can revisit this process, to automate it using https://github.com/ionide/KeepAChangelog.

File: src/Giraffe/Giraffe.fsproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <!-- General -->
     <AssemblyName>Giraffe</AssemblyName>
+    <AssemblyVersion>7.0.2</AssemblyVersion>
```
Comment (64J0) [inline src/Giraffe/Giraffe.fsproj:5]: Another potential fix would be to define this property at the `dotnet pack` command on CI here https://github.com/giraffe-fsharp/Giraffe/blob/master/.github/workflows/publish.yml#L69

**Review by nojaf — APPROVED** [review]

**Review by mrtz-j — APPROVED** [review]

**Comment by kerams** [comment]
Is this causing the assembly file version to be `0.0.0`? The previous releases used `1.0.0`, so Fable.Remoting now throws an exception when used with the v8 alpha.

**Comment by 64J0** [comment]
> Is this causing the assembly file version to be `0.0.0`? The previous releases used `1.0.0`, so Fable.Remoting now throws an exception when used with the v8 alpha.

It must be related. I just noticed that although the `AssemblyFileVersion` is defined properly for the alpha release (https://nuget.info/packages/Giraffe/8.0.0-alpha-001):

image

The version value at the Full Name is not.

This is what we had before:

image

**Comment by 64J0** [comment]
I think this must be enough to fix this problem @kerams: https://github.com/giraffe-fsharp/Giraffe/pull/655. Once this is fixed, I'm going to generate a new alpha release.

**Comment by 64J0** [comment]
Now it's showing something reasonable https://nuget.info/packages/Giraffe/8.0.0-alpha-002.

image

**Comment by kerams** [comment]
Yeah, it works, thank you.

**Pull request merged**